src/PlusDataCollection/Atracsys/vtkPlusAtracsysTracker.cxx

```diff
@@ -0,0 +1,396 @@
+/*=Plus=header=begin======================================================
+Program: Plus
+Copyright (c) Laboratory for Percutaneous Surgery. All rights reserved.
+See License.txt for details.
+=========================================================Plus=header=end*/
+
+// Local includes
+#include "PlusConfigure.h"
+#include "vtkPlusAtracsysTracker.h"
+#include "AtracsysTracker.h"
+
+// VTK includes
+#include <vtkSmartPointer.h>
+#include <vtkMatrix4x4.h>
+#include <vtkMath.h>
+
+// System includes
+#include <string>
+#include <map>
+#include <fstream>
+#include <iostream>
+
+#ifdef __linux__
+#include <unistd.h>
+#endif
+#ifdef WIN32
+#include <windows.h>
+#endif
+
+// Atracsys includes
+#include "ftkErrors.h"
+#include "ftkEvent.h"
+#include "ftkInterface.h"
+#include "ftkOptions.h"
+#include "ftkPlatform.h"
+#include "ftkTypes.h"
+
+vtkStandardNewMacro(vtkPlusAtracsysTracker);
+
+// for convenience
+#define ATR_SUCCESS AtracsysTracker::ATRACSYS_RESULT::SUCCESS
+typedef AtracsysTracker::ATRACSYS_RESULT ATRACSYS_RESULT;
+
+//----------------------------------------------------------------------------
+class vtkPlusAtracsysTracker::vtkInternal
+{
+public:
+  vtkPlusAtracsysTracker* External;
+
+  vtkInternal(vtkPlusAtracsysTracker* external)
+    : External(external)
+  {
+  }
+
+  virtual ~vtkInternal()
+  {
+  }
+
+  int MaxMissingFiducials = 0;
+  float MaxMeanRegistrationErrorMm = 2.0;
+  int ActiveMarkerPairingTimeSec = 0;
+
+  // type of tracker connected
+  AtracsysTracker::DEVICE_TYPE DeviceType = AtracsysTracker::UNKNOWN_DEVICE;
+
+  // matches plus tool id to .ini geometry file names/paths
+  std::map<std::string, std::string> PlusIdMappedToGeometryFilename;
+
+  // matches fusionTrack internal tool geometry ID to Plus tool ID for updating tools
+  std::map<int, std::string> FtkGeometryIdMappedToToolId;
+
+  // Atracsys API wrapper class handle
+  AtracsysTracker* Tracker = new AtracsysTracker();
```

Comment:
Why not simply AtracsysTracker Tracker?
If we do need to use new operator then it has to be deleted in the destructor.